api/src/services/users.js

```diff
@@ -551,16 +551,21 @@ const validateUserContact = (data, user, userSettings) => {
   }
 };
 
-const createUserEntities = async (user, appUrl, preservePrimaryContact) => {
+const createUserEntities = async (data, appUrl) => {
+  // preserve the place's primary contact only if it's an existing one
+  // => if we're creating the place alongside the user, set the contact as the place's primary contact
+  const isCreatingNewPlace = !_.isUndefined(data.place) && !_.isNull(data.place) && _.isUndefined(data.place._id);
```

Comment:
We're not using this variable much right? And since we already have a comment explaining the business logic.
What about if we just keep the `preservePrimaryContact`? 

```
preservePrimaryContact = !!(data && data.place && data.place._id)
```